src/we_random.c

```diff
@@ -346,8 +290,58 @@ static int we_rand_pseudorand(unsigned char *buf, int num)
         /* Mix global seed if RAND_add() or RAND_seed() has been called. */
         if (haveSeed) {
             ret = we_rand_mix_seed(buf, num, we_seed, sizeof(we_seed));
+            if (ret != 1) {
+                WOLFENGINE_ERROR_MSG(WE_LOG_RNG, "we_rand_mix_seed with global "
+                                                 "seed failed");
+            }
         }
-    #endif
+        /* Mix in weak entropy: thread ID, current time, PID. */
+        if (ret == 1) {
+            threadId = CRYPTO_thread_id();
```

Comment:
Put all the data (pids and time) into one buffer and mix that into returned buffer. Number of bytes could be very long and mixing multiple times will be expensive.